src/main/java/org/opensearch/security/tools/democonfig/Installer.java

```diff
@@ -0,0 +1,377 @@
+    static String OPENSEARCH_BIN_DIR;
+    static String OPENSEARCH_PLUGINS_DIR;
+    static String OPENSEARCH_LIB_PATH;
+    static String OPENSEARCH_INSTALL_TYPE;
+    static String OPENSEARCH_CONF_DIR;
+    static String OPENSEARCH_VERSION;
+    static String SECURITY_VERSION;
+
+    static ExecutionEnvironment environment = ExecutionEnvironment.demo;
+
+    static final String OS = System.getProperty("os.name") + " " + System.getProperty("os.version") + " " + System.getProperty("os.arch");
+
+    static final String FILE_EXTENSION = OS.toLowerCase().contains("win") ? ".bat" : ".sh";
+
+    static final String SYSTEM_INDICES = ".plugins-ml-config, .plugins-ml-connector, .plugins-ml-model-group, .plugins-ml-model, "
+        + ".plugins-ml-task, .plugins-ml-conversation-meta, .plugins-ml-conversation-interactions, .opendistro-alerting-config, .opendistro-alerting-alert*, "
+        + ".opendistro-anomaly-results*, .opendistro-anomaly-detector*, .opendistro-anomaly-checkpoints, .opendistro-anomaly-detection-state, "
+        + ".opendistro-reports-*, .opensearch-notifications-*, .opensearch-notebooks, .opensearch-observability, .ql-datasources, "
+        + ".opendistro-asynchronous-search-response*, .replication-metadata-store, .opensearch-knn-models, .geospatial-ip2geo-data*";
+
+    static SecuritySettingsConfigurer securitySettingsConfigurer;
+    static CertificateGenerator certificateGenerator;
+
+    public static void main(String[] options) {
+        securitySettingsConfigurer = new SecuritySettingsConfigurer();
+        certificateGenerator = new CertificateGenerator();
+
+        printScriptHeaders();
+        readOptions(options);
+        gatherUserInputs();
+        initializeVariables();
+        printVariables();
+        securitySettingsConfigurer.configureSecuritySettings();
+        certificateGenerator.createDemoCertificates();
+        finishScriptExecution();
+    }
+
+    /**
+     * Prints deprecation warning and other headers for the script
+     */
+    static void printScriptHeaders() {
+        System.out.println("**************************************************************************");
+        System.out.println("** This tool will be deprecated in the next major release of OpenSearch **");
+        System.out.println("** https://github.com/opensearch-project/security/issues/1755           **");
+        System.out.println("**************************************************************************");
+        System.out.println("\n");
+        System.out.println("### OpenSearch Security Demo Installer");
+        System.out.println("### ** Warning: Do not use on production or public reachable systems **");
+    }
+
+    /**
+     * Reads the options passed to the script
+     * @param options an array of strings containing options passed to the script
+     */
+    static void readOptions(String[] options) {
+        // set script execution dir
+        SCRIPT_DIR = options[0];
```

Comment:
Should we specify that we need to pass the script directory now? I don't see it mentioned in the help, but I don't recall having to pass the path before. This seems like something we should make very explicit unless you have a way of autopopulating it (I have not read all the files yet).

Reply:
BASE_DIR always used script path. Take a look here: https://github.com/opensearch-project/security/blob/b7f47b1f516bee03a7dfaeec3a9133e4eafa3a73/tools/install_demo_configuration.sh#L101